src/components/Admin/AddNewUserDialog.js

```diff
@@ -0,0 +1,142 @@
+import React, { useState } from 'react';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { TextField, Button, Select, MenuItem, FormControl, InputLabel } from '@material-ui/core';
+import axios from 'axios';
+
+export default function AddNewUser({ isOpen, onClose, roles }) {
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [role, setRole] = useState(0);
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [errors, setErrors] = useState({
+    firstName: '',
+    lastName: '',
+    email: '',
+    password: '',
+  });
+
+  const handleAddNewUser = () => {
+    let error = { ...errors };
+
+    error.firstName = firstName !== '' ? '' : 'no empty First Name';
+    error.lastName = lastName !== '' ? '' : 'no empty Last Name';
+    error.email = email !== '' ? '' : 'no empty Email';
+    error.password = password !== '' ? '' : 'no empty Password';
+
+    setErrors(error);
+
+    if (
+      Object.values(error).reduce((sum, err) => {
+        return sum + err.length;
+      }, 0) > 0
+    ) {
+      return;
+    }
+
+    axios
```

Comment:
create a separate service to do rest requests with methods get, post, put, delete and reuse it here

Reply:
Done, check Axios.js in src/components.